### PR TITLE
MariadbConnector support for m1 macs

### PR DESCRIPTION
**Pull request by lucyllewy**
Followup to #224

* Drop use of `CMAKE_OSX_ARCHITECTURES`
* Update OpenSSL to version 1.1.1k for m1 configure script enablement
* Add `arm64` branch to the apple-specific configure commands setup

### CODE REVIEW

**Comment by Simn** [comment]
#224 caused conflicts here which will have to be resolved.

File: libs/mysql/CMakeLists.txt [hunk 1]
```diff
@@ -4,10 +4,12 @@
 
 if (STATIC_OPENSSL)
 	if (APPLE)
-		if (${CMAKE_OSX_ARCHITECTURES} STREQUAL "i386")
-			set(OPENSSL_CONF ./Configure darwin-i386-cc "-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
-		elseif (${CMAKE_OSX_ARCHITECTURES} STREQUAL "x86_64")
+		if (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i386")
+			set(OPENSSL_CONF ./Configure $CMAKE_C_COMPILER "-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
```
Comment (tobil4sk) [inline libs/mysql/CMakeLists.txt:8]: Is there a reason to use `$CMAKE_C_COMPILER` just for this line? If so, shouldn't it be `${CMAKE_C_COMPILER}`?

**Comment by tobil4sk** [comment]
> https://github.com/HaxeFoundation/neko/pull/224 caused conflicts here which will have to be resolved.

To hopefully speed things up, I've resolved the conflicts in this branch: https://github.com/tobil4sk/neko/commits/m1-mysql.

**Comment by Simn** [comment]
I just made you maintainer (thought I already did that, sorry!), you should be able to push to PR branches now.

**Pull request merged**